protocols/imap/src/main/java/org/apache/james/imap/decode/ImapRequestLineReader.java

```diff
@@ -814,6 +815,24 @@ private long parseUnsignedInteger(String value) throws DecodingException {
         boolean isValid(char chr);
     }
 
+    public static class StringValidator implements ImapRequestLineReader.CharacterValidator {
```

Comment:
The behavior of this class is confusing enough to deserve some documentation (or if it's possible better naming)